tests/testthat/test-tab_afun_cfun.R

```diff
@@ -0,0 +1,233 @@
+        split_cols_by("ARMCD") %>% 
+        analyze("BMRKR1", afun = afun_tmp)
+    lyt_row <- basic_table() %>% 
+        split_rows_by("STRATA1", split_fun = keep_split_levels("A")) %>% 
+        split_rows_by("ARMCD") %>% 
+        analyze("BMRKR1", afun = afun_tmp)
+    
+    # Error on the columns should happen because it is used for counts on column space
+    expect_error(lyt_col %>% build_table(ex_adsl, alt_counts_df = DM))
+    expect_silent(lyt_row %>% build_table(ex_adsl, alt_counts_df = DM))
+})
+
+context("Content functions (cfun)")
+
+test_that(".alt_df appears in cfun but not in afun.", {
+    # Adding STRATA2 col to DM for alt_counts_df col split
+    alt_tmp <- DM %>% left_join(ex_adsl %>% 
+                                    mutate(ID = paste0("S", seq_len(nrow(ex_adsl)))) %>% 
+                                    select(ID, STRATA2))
+    
+    afun_tmp <- function(x, ...) rcell(mean(x), label = "MeAn", format = "xx.x")
+    cfun_tmp <- function(x, .alt_df, labelstr, .N_col, 
+                         .spl_context,
+                         .all_col_exprs,
+                         .all_col_counts,
+                         ...) {
+        if (!missing(.alt_df)) {
+            # .alt_df is only row splitted matter
+            stopifnot(nrow(alt_tmp %>% filter(STRATA1 == "A")) == nrow(.alt_df))
+            
+            # Filtered column number of elemens correspond to .N_col
+            stopifnot(nrow(alt_tmp %>% 
+                               filter(eval(.spl_context$cur_col_expr[[1]]))) == .N_col)
+        } else {
+            # Checking cur_col_n is the same as .N_col for root and length(x) for split
+            stopifnot(identical(.spl_context$cur_col_n, c(.N_col, length(x))))
+        }
+        
+        # Checking internal names for all column counts correspond to .spl_context
+        stopifnot(all(names(.all_col_counts) %in% colnames(.spl_context)))
+        
+        # Checking that current col id and col counts agree with .N_col
+        stopifnot(.all_col_counts[.spl_context$cur_col_id[1]] == .N_col)
+        
+        # Checking col expression
+        stopifnot(identical(.all_col_exprs[.spl_context$cur_col_id[1]][[1]],
+                            .spl_context$cur_col_expr[[1]])) # Uses the root one
+        
+        in_rows(c(length(x), length(x) / .N_col), 
+                .names = labelstr, 
+                .formats = c("xx (xx.xx)"))
+    }
+    
+    lyt <- basic_table(show_colcounts = TRUE) %>% 
+        split_cols_by("SEX", split_fun = keep_split_levels(c("F", "U"))) %>% 
+        split_cols_by("STRATA2", split_fun = keep_split_levels("S1")) %>% 
+        split_rows_by("STRATA1", split_fun = keep_split_levels("A")) %>% 
+        summarize_row_groups(var = "BMRKR1", cfun = cfun_tmp) %>%
+        split_rows_by("ARMCD") %>% 
+        analyze("BMRKR1", afun = afun_tmp)
+    
+    expect_silent(lyt %>% build_table(ex_adsl))
+    expect_error(lyt %>% build_table(ex_adsl, alt_counts_df = DM))
```

Comment:
Similar to above, would be nice to improve the error.
```
Error in FUN(X[[i]], ...) : 
  alt_counts_df (or df) appears incompatible with column-split structure. Offending column subset expression: ((!is.na(SEX) & SEX %in% "F")) & (!is.na(STRATA2) & STRATA2 %in% 
Original error message: object 'STRATA2' not foundalt_counts_df (or df) appears incompatible with column-split structure. Offending column subset expression:     "S1")
Original error message: object 'STRATA2' not found
```

Reply:
This looks like a paste error where we are constructing the message but one of the elements in inexpectedly of length `> 1`

Reply:
I did not fix the double error print as it seems a problem unrelated to this PR and we can deal with it at a later stage. Changed the ref to dataset name